### PR TITLE
Check if default response is defined when matching unknown response

### PR DESCRIPTION
**Pull request by shaxbee**

### CODE REVIEW

File: src/validate.ts [hunk 1]
```diff
@@ -178,6 +178,12 @@ export function response(compiledPath: CompiledPath | undefined,
   if (response === undefined) {
     response = operation.responses['default'];
   }
+  if (response === undefined) {
```
Comment (carlansley) [inline src/validate.ts]: Definitely an issue here, thanks for finding it!  Although from my reading of the spec if there's no response defined, no content should be returned.  So maybe there should instead be something at the top of the `validate` function like:
```
 if (schema === undefined) {
  if (!value) {
    return;
  } else {
   return {
      actual: value,
      expected: undefined
    };
  } 
```

Let me know what you think.

Comment (carlansley) [inline src/validate.ts]: Actually on second reading there's really no specified behavior for this situation.  I'll merge this PR but may move the error check to the `validate` function.  Thanks!

**Comment by shaxbee** [comment]
To be honest your solution seems to be more reasonable :-)

**Comment by carlansley** [comment]
:) it may well be but in the absence of clear spec-defined behavior I think an error is a logical approach for now.  swagger2-koa will respond with a 500 which I think is ok; the spec does say that all known responses (good & errors) should be defined.

I'm going to raise an issue on OA/OpenAPI-Specification about this.

**Pull request merged**